oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java

```diff
@@ -119,4 +127,17 @@ private static Consumer<List<String>> clientAuthenticationMethods() {
 	private static String asUrl(String issuer, String endpoint) {
 		return UriComponentsBuilder.fromUriString(issuer).path(endpoint).build().toUriString();
 	}
+
+	private boolean isDynamicClientRegistrationEnabled() {
+		DefaultSecurityFilterChain defaultSecurityFilterChain = (DefaultSecurityFilterChain) WebApplicationContextUtils
+				.getRequiredWebApplicationContext(getServletContext()).getBean("authorizationServerSecurityFilterChain");
```

Comment:
Thanks for the updates @sahariardev.

We cannot rely on the bean being called `authorizationServerSecurityFilterChain`. The consuming application could name the bean to something different and then this logic would not work.

Reply:
@sahariardev Sorry for the delayed response. I haven't had a chance to revisit this yet. I have some re-factoring tasks that are higher priority at the moment. I will touch base as soon as I get a chance to revisit this.

Reply:
Okay, no problem.

Reply:
Thanks for your patience @sahariardev. I just merged gh-616 via 0994a1e1e14263d106373e3397f8c5c44f9a7708.

Based on the updates in that commit, the following code (add to `OidcConfigurer.configure()`) will allow you to determine when `oidc.clientRegistrationEndpoint()` is enabled so you can apply the `registration_endpoint` configuration:

```java
OidcClientRegistrationEndpointConfigurer clientRegistrationEndpointConfigurer =
		getConfigurer(OidcClientRegistrationEndpointConfigurer.class);
if (clientRegistrationEndpointConfigurer != null) {
	OidcProviderConfigurationEndpointConfigurer providerConfigurationEndpointConfigurer =
			getConfigurer(OidcProviderConfigurationEndpointConfigurer.class);
	providerConfigurationEndpointConfigurer.addDefaultProviderConfigurationCustomizer(
			builder -> builder.clientRegistrationEndpoint("https://localhost/connect/register"));
}
```

Let me know if you have any questions. Thanks.

Reply:
Also, please rebase the PR off of `0.4.x` branch.

Reply:
Thanks @jgrandja. I will update those ASAP.

Reply:
Hi  @jgrandja,
I have started working on this. As I was facing some trouble to rebase with 0.4.x, I had do close this one and opening a new PR. Will let you know the updates ASAP.

Reply:
Here is the new PR link. let's continue our discussion there.